src/DataMapper/Product/Provider/ReflectionAttributeValuesProvider.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusMeilisearchPlugin\DataMapper\Product\Provider;
+
+use Setono\SyliusMeilisearchPlugin\Document\Document;
+use Webmozart\Assert\Assert;
+
+final class ReflectionAttributeValuesProvider implements ReflectionAttributeValuesProviderInterface
```

Comment:
I believe this code belongs on the `Metadata`, both for consistency and performance